klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+# Copyright 2022 GlobalFoundries PDK Authors
```

Comment:
looks like flake8 is ignoring most of the violations:
```
ignore = E203, E266, E501, W503, F403, F401
```
when we turn those on we get:
```
./klayout/drc/testing/run_regression.py:30:1: F401 'subprocess.Popen' imported but unused
./klayout/drc/testing/run_regression.py:30:1: F401 'subprocess.PIPE' imported but unused
./klayout/drc/testing/run_regression.py:61:89: E501 line too long (90 > 88 characters)
./klayout/drc/testing/run_regression.py:85:89: E501 line too long (108 > 88 characters)
./klayout/drc/testing/run_regression.py:97:89: E501 line too long (91 > 88 characters)
./klayout/drc/testing/run_regression.py:119:89: E501 line too long (105 > 88 characters)
./klayout/drc/testing/run_regression.py:123:34: E203 whitespace before ':'
./klayout/drc/testing/run_regression.py:126:89: E501 line too long (105 > 88 characters)
./klayout/drc/testing/run_regression.py:216:89: E501 line too long (99 > 88 characters)
./klayout/drc/testing/run_regression.py:235:89: E501 line too long (168 > 88 characters)
./klayout/drc/testing/run_regression.py:256:89: E501 line too long (100 > 88 characters)
./klayout/drc/testing/run_regression.py:264:89: E501 line too long (109 > 88 characters)
./klayout/drc/testing/run_regression.py:320:89: E501 line too long (119 > 88 characters)
./klayout/drc/testing/run_regression.py:321:89: E501 line too long (112 > 88 characters)
./klayout/drc/testing/run_regression.py:322:89: E501 line too long (107 > 88 characters)
./klayout/drc/testing/run_regression.py:330:89: E501 line too long (90 > 88 characters)
./klayout/drc/testing/run_regression.py:332:89: E501 line too long (109 > 88 characters)
./klayout/drc/testing/run_regression.py:338:89: E501 line too long (95 > 88 characters)
./klayout/drc/testing/run_regression.py:339:89: E501 line too long (97 > 88 characters)
./klayout/drc/testing/run_regression.py:380:89: E501 line too long (91 > 88 characters)
./klayout/drc/testing/run_regression.py:382:89: E501 line too long (93 > 88 characters)
./klayout/drc/testing/run_regression.py:394:89: E501 line too long (90 > 88 characters)
./klayout/drc/testing/run_regression.py:463:89: E501 line too long (106 > 88 characters)
./klayout/drc/testing/run_regression.py:513:89: E501 line too long (96 > 88 characters)
./klayout/drc/testing/run_regression.py:518:89: E501 line too long (96 > 88 characters)
./klayout/drc/testing/run_regression.py:523:89: E501 line too long (96 > 88 characters)
./klayout/drc/testing/run_regression.py:622:9: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:628:9: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:644:89: E501 line too long (143 > 88 characters)
./klayout/drc/testing/run_regression.py:647:89: E501 line too long (254 > 88 characters)
./klayout/drc/testing/run_regression.py:650:89: E501 line too long (193 > 88 characters)
./klayout/drc/testing/run_regression.py:653:89: E501 line too long (316 > 88 characters)
./klayout/drc/testing/run_regression.py:666:89: E501 line too long (143 > 88 characters)
./klayout/drc/testing/run_regression.py:669:89: E501 line too long (254 > 88 characters)
./klayout/drc/testing/run_regression.py:683:89: E501 line too long (102 > 88 characters)
./klayout/drc/testing/run_regression.py:721:89: E501 line too long (93 > 88 characters)
./klayout/drc/testing/run_regression.py:732:89: E501 line too long (94 > 88 characters)
./klayout/drc/testing/run_regression.py:761:89: E501 line too long (91 > 88 characters)
./klayout/drc/testing/run_regression.py:785:89: E501 line too long (95 > 88 characters)
./klayout/drc/testing/run_regression.py:788:5: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:790:89: E501 line too long (90 > 88 characters)
./klayout/drc/testing/run_regression.py:793:5: E266 too many leading '#' for block comment
flake8.main.application   MainProcess    163 INFO     Found a total of 105 violations and reported 105
./klayout/drc/testing/run_regression.py:800:5: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:805:5: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:810:5: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:815:5: E266 too many leading '#' for block comment
./klayout/drc/testing/run_regression.py:846:89: E501 line too long (95 > 88 characters)
./klayout/drc/testing/run_regression.py:865:5: E266 too many leading '#' for block comment
```
I think we may want to turn those on? wdyt?

Reply:
May be F401 is important but the others I don't think they are that critical. If you want me to remove those limitations, no problem, I'll do and clean the code. Please let me know what you have on your mind?

Reply:
maybe then log an issue that we should remove those? so that we can address it in a separate CL.

Things like https://pypi.org/project/autopep8/ or https://pypi.org/project/black/ can help ensuring consistent style over a codebase.

Reply:
Done and created a new issue https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/issues/4

Reply:
@proppy BTW, I use @joamatab recommendations on that.